src/mqueryfront/src/QueryTimer.js

```diff
@@ -0,0 +1,62 @@
+import React, { Component } from "react";
+
+class QueryTimer extends Component {
+    constructor(props) {
+        super(props);
+        this.state = { currentTime: 0 };
+    }
+
+    tick() {
+        this.setState({
+            currentTime: Math.floor(Date.now() / 1000),
+        });
+    }
+
+    componentDidMount() {
+        this.interval = setInterval(() => this.tick(), 1000);
+    }
+
+    componentWillUnmount() {
+        clearInterval(this.interval);
+    }
+
+    render() {
+        if (
+            !this.props.job.submitted ||
+            ["done", "cancelled", "failed", "expired"].includes(
```

Comment:
duplicated with `QueryPage.js` and `QueryResultsStatus.js` => move the array to some shared common file.